src/components/Chat/hooks/useAppStateListener.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useRef } from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+
+import type { StreamChat } from 'stream-chat';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+/**
+ * Disconnect the websocket connection when app goes to background,
+ * and reconnect when app comes to foreground.
+ * We do this to make sure, user receives push notifications when app is in the background.
+ * You can't receive push notification until you have active websocket connection.
+ */
+export const useAppStateListener = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  client: StreamChat<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const appState = useRef(AppState.currentState);
+
+  useEffect(() => {
+    AppState.addEventListener('change', handleAppStateChange);
+
+    return () => {
+      AppState.removeEventListener('change', handleAppStateChange);
+    };
+  }, []);
+
+  const handleAppStateChange = async (nextAppState: AppStateStatus) => {
```

Comment:
This will be an issue; if you get a phone call lets say, the app state will go from active -> inactive (no connection closed), then you end call and in app, app state will go from inactive -> active (new connection), now 2x connections.

Reply:
This is now handled on js client level :)

Reply:
we should still get the logic right tho? I.e. if we only close when going to background only open when coming from background??

Reply:
If app is in inactive state for too long, websocket may still be dropped. So I think it's just better to call openConnection anyways.

Reply:
Actually curious if that happens, do you know?